### ISSUE TITLE
menu button suspends tvOS app when TVEventControl.disableGestureHandlersCancelTouches is used

### ISSUE THREAD

**Issue by mateusz1913** [react-native-tvos/react-native-tvos#678]
### Description

In a project we use `UISearchController` as a native component. It works when tap and long press gesture recognizers in `RCTTVRemoteHandler` class have `cancelsTouchesInView` property set to `NO`. That can be controled from JS codebase via `TVEventControl.[disable|enable]GestureHandlersCancelTouches` API and in our case we want to use `TVEventControl.disableGestureHandlersCancelTouches` globally.

However the side effect is that with current implementation of `RCTTVRemoteHandler`, the menu button emits event for both `RCTTVRemoteHandler` and default system recognizers. That means that when I press back button on tvOS remote in an application, the app will consume the event via `BackHandler` API, but simultaneously the app is suspended.

I found [that comment](https://github.com/react-native-tvos/react-native-tvos/pull/366#issuecomment-1282946019) and it seems that having the UIKit component with some gestures like `UISearchController` rendered inside of react tree is in fact a use case when menu button should always have `cancelsTouchesInView` set to `YES` 😄 

The proposed fix would be to **not** set the `cancelsTouchesInView` for menu button recognizer inside of `RCTTVRemoteHandler.m`

```diff
 - (void)enableGestureHandlersCancelTouches
{
  dispatch_async(dispatch_get_main_queue(), ^{
    for (NSString *name in [self.tvRemoteGestureRecognizers allKeys]) {
      self.tvRemoteGestureRecognizers[name].cancelsTouchesInView = YES;
    }
-    [self.tvMenuKeyRecognizer setCancelsTouchesInView:YES];
    [self.tvPanGestureRecognizer setCancelsTouchesInView:YES];
  });
}

 - (void)disableGestureHandlersCancelTouches
{
  dispatch_async(dispatch_get_main_queue(), ^{
    for (NSString *name in [self.tvRemoteGestureRecognizers allKeys]) {
      self.tvRemoteGestureRecognizers[name].cancelsTouchesInView = NO;
    }
-    [self.tvMenuKeyRecognizer setCancelsTouchesInView:NO];
    [self.tvPanGestureRecognizer setCancelsTouchesInView:NO];
  });
}
```

@douglowder @yusufyildirim let us know, if you're OK with that idea and if yes, I'll provide a PR

### React Native Version

0.73.4-0

### Output of `npx react-native info`

```
System:
  OS: macOS 14.2.1
  CPU: (10) arm64 Apple M1 Pro
  Memory: 470.02 MB / 32.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.5.0
    path: ~/.nvm/versions/node/v20.5.0/bin/node
  Yarn:
    version: 4.0.2
    path: ~/.nvm/versions/node/v20.5.0/bin/yarn
  npm:
    version: 9.8.0
    path: ~/.nvm/versions/node/v20.5.0/bin/npm
  Watchman:
    version: 2023.06.26.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.13.0
    path: /Users/mateuszmedrek/.rbenv/shims/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.2
      - iOS 17.2
      - macOS 14.2
      - tvOS 17.2
      - visionOS 1.0
      - watchOS 10.2
  Android SDK:
    API Levels:
      - "31"
      - "33"
      - "34"
    Build Tools:
      - 30.0.3
      - 31.0.0
      - 33.0.0
      - 33.0.1
      - 34.0.0
    System Images:
      - android-33 | Google TV ARM 64 v8a
      - android-33 | Google APIs ARM 64 v8a
      - android-34 | Google APIs ARM 64 v8a
    Android NDK: Not Found
IDEs:
  Android Studio: 2023.2 AI-232.10227.8.2321.11479570
  Xcode:
    version: 15.2/15C500b
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.7
    path: /Users/mateuszmedrek/.jenv/shims/javac
  Ruby:
    version: 2.7.5
    path: /Users/mateuszmedrek/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react: Not Found
  react-native: Not Found
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false
```

### Steps to reproduce

1. Change `TVEventControl.enableGestureHandlersCancelTouches();` to `TVEventControl.disableGestureHandlersCancelTouches();` inside of `RNTesterAppShared.js` in RNTester
2. Run RNTester app
3. Navigate to one of examples
4. Navigate back with menu button

Result: app navigated back to screen with list of examples and become suspended

### Snack, screenshot, or link to a repository

I am attaching the recording of RNTester app with `TVEventControl.disableGestureHandlersCancelTouches();` used inside of `RNTesterAppShared.js` in RNTester

https://github.com/react-native-tvos/react-native-tvos/assets/25980166/487ed20a-9f3b-456e-a66d-dc8e3b02462e

**Comment by douglowder** [comment]
Hmm for features like this, we may need more fine grained control over the gesture handlers... I'll take a look.

**Comment by douglowder** [comment]
Fixed in #699
